pandas/core/arrays/string_.py

```diff
@@ -279,11 +268,27 @@ def __setitem__(self, key, value):
                 value = np.asarray(value, dtype=object)
             if len(value) and not lib.is_string_array(value, skipna=True):
                 raise ValueError("Must provide strings.")
+        return value
+
+    def __setitem__(self, key, value):
+        value = extract_array(value, extract_numpy=True)
+        if isinstance(value, type(self)):
+            # extract_array doesn't extract PandasArray subclasses
+            value = value._ndarray
+
+        key = check_array_indexer(self, key)
+        scalar_key = lib.is_scalar(key)
+        scalar_value = lib.is_scalar(value)
+        if scalar_key and not scalar_value:
+            raise ValueError("setting an array element with a sequence.")
+
+        value = self._validate_setitem_value(value)
 
         super().__setitem__(key, value)
 
     def fillna(self, value=None, method=None, limit=None):
-        # TODO: validate dtype
+        if value is not None:
+            value = self._validate_setitem_value(value)
```

Comment:
This is actually not needed, as the setitem under the hood of fillna already calls it. So calling it here as well means we validate the value twice.

Reply:
right, reverted all the changes here